### PR TITLE
New deposit

### PR DESCRIPTION
**Pull request by enriquesouza**

### CODE REVIEW

File: VultisigApp/VultisigApp/Chains/dydx.swift [hunk 1]
```diff
@@ -20,7 +21,7 @@ class DydxHelper {
     static let DydxGasLimit:UInt64 = 2500000000000000
     
     func getSwapPreSignedInputData(keysignPayload: KeysignPayload,signingInput: CosmosSigningInput) -> Result<Data,Error> {
-        guard case .Cosmos(let accountNumber, let sequence,let gas) = keysignPayload.chainSpecific else {
+        guard case .Cosmos(let accountNumber, let sequence,let gas, let isDeposit) = keysignPayload.chainSpecific else {
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Chains/dydx.swift:24]: shouldn't here also be `transactionType`?

File: VultisigApp/VultisigApp/Chains/atom.swift [hunk 2]
```diff
@@ -20,7 +20,7 @@ class ATOMHelper {
     static let ATOMGasLimit:UInt64 = 200000
  
     func getSwapPreSignedInputData(keysignPayload: KeysignPayload, signingInput: CosmosSigningInput) throws -> Data {
-        guard case .Cosmos(let accountNumber, let sequence,let gas) = keysignPayload.chainSpecific else {
+        guard case .Cosmos(let accountNumber, let sequence,let gas, let isDeposit) = keysignPayload.chainSpecific else {
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Chains/atom.swift:23]: transactionType?

File: VultisigApp/VultisigApp/Chains/dydx.swift (hunk 1 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Chains/dydx.swift:24]: I’ll double check, tks

File: VultisigApp/VultisigApp/Chains/atom.swift (hunk 2 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Chains/atom.swift:23]: I’ll double check, tks

File: VultisigApp/VultisigApp/Chains/kujira.swift [hunk 3]
```diff
@@ -18,9 +18,9 @@ class KujiraHelper {
     }
     
     static let kujiraGasLimit:UInt64 = 200000
-  
+    
     func getSwapPreSignedInputData(keysignPayload: KeysignPayload,signingInput: CosmosSigningInput) throws -> Data {
-        guard case .Cosmos(let accountNumber, let sequence,let gas) = keysignPayload.chainSpecific else {
+        guard case .Cosmos(let accountNumber, let sequence, let gas, let isDeposit) = keysignPayload.chainSpecific else {
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Chains/kujira.swift:23]: can you please clean up this as well?

File: VultisigApp/VultisigApp/Chains/kujira.swift [hunk 4]
```diff
@@ -43,9 +43,9 @@ class KujiraHelper {
         // deposit message has been set
         return try input.serializedData()
     }
-
+    
     func getPreSignedInputData(keysignPayload: KeysignPayload) throws -> Data {
-        guard case .Cosmos(let accountNumber, let sequence , let gas) = keysignPayload.chainSpecific else {
+        guard case .Cosmos(let accountNumber, let sequence , let gas, let isDeposit) = keysignPayload.chainSpecific else {
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Chains/kujira.swift:48]: same here, `isDeposit` is not used , and also it is accurate

**Review by johnnyluo — CHANGES_REQUESTED** [review]

File: VultisigApp/VultisigApp/Chains/kujira.swift (hunk 4 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Chains/kujira.swift]: Sure. Done.

File: VultisigApp/VultisigApp/Chains/kujira.swift (hunk 3 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Chains/kujira.swift]: Done

**Review by johnnyluo — APPROVED** [review]
LGTM

**Pull request merged**